pkg/gatherers/clusterconfig/machine_configs.go

```diff
@@ -0,0 +1,101 @@
+package clusterconfig
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+var (
+	sapMachineConfigNameList = map[string]struct{}{
+		"75-worker-sap-data-intelligence": {},
+		"75-master-sap-data-intelligence": {},
+	}
+)
+
+// GatherMachineConfigs collects MachineConfigs definitions. Following data is intentionally removed from the definitions:
+// * `spec.config.storage.files`
+// * `spec.config.passwd.users`
+//
+// The SDI/SAP related MachineConfigs are an exception and are gathered completely
+//
+// Response see https://docs.openshift.com/container-platform/4.7/rest_api/machine_apis/machineconfig-machineconfiguration-openshift-io-v1.html
+//
+// * Location in archive: config/machineconfigs/<name>.json
+// * Id in config: machine_configs
+// * Since versions:
+//   * 4.8+
+func (g *Gatherer) GatherMachineConfigs(ctx context.Context) ([]record.Record, []error) {
+	gatherDynamicClient, err := dynamic.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return gatherMachineConfigs(ctx, gatherDynamicClient)
+}
+
+func isSAPMachineConfig(mc unstructured.Unstructured) bool {
+	if _, exists := sapMachineConfigNameList[mc.GetName()]; exists {
+		return true
+	}
+
+	for labelName, labelValue := range mc.GetLabels() {
+		if labelName == "workload" && labelValue == "sap-data-intelligence" {
+			return true
+		}
+	}
+
+	for _, ownerRef := range mc.GetOwnerReferences() {
+		if ownerRef.Kind == "ContainerRuntimeConfig" && ownerRef.Name == "sdi-pids-limit" {
+			return true
+		}
+	}
+
+	return false
+}
+
+func gatherMachineConfigs(ctx context.Context, dynamicClient dynamic.Interface) ([]record.Record, []error) {
+	mcList, err := dynamicClient.Resource(machineConfigGroupVersionResource).List(ctx, metav1.ListOptions{})
+	if errors.IsNotFound(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	records := []record.Record{}
+	var errs []error
+	for i := range mcList.Items {
+		mc := mcList.Items[i]
+		// preservce the original behavior - that is gathering of complete SAP machineconfig definitions
+		if !isSAPMachineConfig(mc) {
```

Comment:
@natiiix Do we need to preserve this logic? I would say no, because the gatherer wasn't called anyway (see the update in `pkg/gatherers/clusterconfig/clusterconfig_gatherer.go`) and more importantly wasn't released in any production version AFAIK.

Reply:
I agree that it most likely didn't make its way into production yet and it's completely fine by me if we remove this part.